### PR TITLE
Suggest method for auto-complete

### PR DESCRIPTION
**Pull request by githubmo**
### What

- Added endpoint for auto-complete
- Using our own snowball analyzer instead of default one with elastic search's suggest

### How to review

Check out the new `/suggest?q=<search>` out 😎

### CODE REVIEW

**Comment by CarlHembrough** [comment]
consider adding some info to the readme to document the new endpoint

**Comment by CarlHembrough** [comment]
run go format

File: handler/search_test.go [hunk 1]
```diff
@@ -141,4 +146,108 @@ func Test(t *testing.T) {
...
+
+		recorder := httptest.NewRecorder()
+		requestBodyReader := bytes.NewReader([]byte("{not a valid document}"))
+		request, _ := http.NewRequest("GET", "/search?q=armed", requestBodyReader)
+
+		Convey("When the query handler is called", func() {
+
+			mockSearchClient := searchtest.NewMockSearchClient()
+			handler.SearchClient = mockSearchClient
+			handler.Search(recorder, request)
```
Comment (CarlHembrough) [inline handler/search_test.go]: This should be calling handler.Suggest() ?

File: handler/search_test.go [hunk 2]
```diff
@@ -141,4 +146,108 @@ func Test(t *testing.T) {
...
+
+			Convey("Then the response contains the content returned from the search client.", func() {
+				So(recorder.Code, ShouldEqual, http.StatusOK)
+
+				actualResponse := &model.SearchResponse{}
+				_ = json.Unmarshal(recorder.Body.Bytes(), actualResponse)
+
+				So(actualResponse.TotalResults, ShouldEqual, mockSearchResponse.TotalResults)
+				So(actualResponse.Results[0].ID, ShouldEqual, mockSearchResponse.Results[0].ID)
+				So(actualResponse.Results[1].ID, ShouldEqual, mockSearchResponse.Results[1].ID)
```
Comment (CarlHembrough) [inline handler/search_test.go]: This should be comparing against the mockSuggestResponse - although they are currently the same 😄

File: handler/search_test.go [hunk 3]
```diff
@@ -141,4 +146,108 @@ func Test(t *testing.T) {
...
+		request, _ := http.NewRequest("GET", "/suggst?q=armed", requestBodyReader)
+
+		Convey("When the query handler is called and the search client returns an error", func() {
+
+			mockSearchClient := searchtest.NewMockSearchClient()
+			mockSearchClient.CustomQueryFunc = func(term string, index string) (*model.SearchResponse, error) {
+				return nil, errors.New("search client error")
+			}
+			handler.SearchClient = mockSearchClient
+			handler.Search(recorder, request)
```
Comment (CarlHembrough) [inline handler/search_test.go]: This should be calling handler.Suggest()?

File: handler/search_test.go [hunk 4]
```diff
@@ -141,4 +146,108 @@ func Test(t *testing.T) {
...
+
+				actualResponse := &model.SearchResponse{}
+				_ = json.Unmarshal(recorder.Body.Bytes(), actualResponse)
+
+				So(actualResponse.TotalResults, ShouldEqual, 0)
+			})
+		})
+	})
+
+	Convey("Given a suggest request", t, func() {
```
Comment (CarlHembrough) [inline handler/search_test.go]: I don't think test test is required for testing the handler.Suggest function as it does not search areas.

Comment (githubmo) [reply handler/search_test.go]: But I'm still testing to see the `Result` is returned? I think this test is still needed.

**Review by CarlHembrough — APPROVED** [review]

**Pull request merged**